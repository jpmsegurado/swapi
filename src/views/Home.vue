<template>
  <div class="home">
    <h1 class="home__title text-center">
      Star Wars App
    </h1>

    <el-row :gutter="20">
      <el-col
        :lg="6"
        :md="8"
        :sm="12"
        :xs="24"
        v-for="person in people"
        :key="person.url"
        class="home__person"
      >
        <person-card
          :name="person.name"
          :gender="person.gender"
          :height="person.height"
          :mass="person.mass"
        />
      </el-col>
    </el-row>

    <div class="home__pagination">
      <el-pagination
        :page-size="params.size"
        :page-count="params.currentPage"
        :total="params.count"
        background
        clas="home__pagination"
        layout="prev, pager, next"
      />
    </div>
  </div>
</template>

<script>
import People from '../api/people'
import PersonCard from '@/components/PersonCard.vue'

export default {
  name: 'Home',
  components: {
    PersonCard
  },
  data () {
    return {
      people: [],
      params: {
        size: 10,
        count: 100,
        currentPage: 2
      }
    }
  },
  async mounted () {
    const { results } = await People.getAll({ page: 1 })
    this.people = results
  }
}
</script>

<style lang="scss" scoped>
.home {
  &__title {
    margin-bottom: 40px;
  }

  &__person {
    margin-bottom: 20px;
  }

  &__pagination {
    display: flex;
    justify-content: center;
  }
}
</style>
