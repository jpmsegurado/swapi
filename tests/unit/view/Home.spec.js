import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

const $route = {
  query: {}
}

function setup () {
  return shallowMount(Home, {
    stubs: {
      'el-row': true,
      'el-col': true,
      'el-button': true,
      'el-input': true,
      'el-pagination': true
    },
    mocks: {
      $route
    }
  })
}

describe('Home', () => {
  describe('#render', () => {
    it('renders correctly', () => {
      const wrapper = setup()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
