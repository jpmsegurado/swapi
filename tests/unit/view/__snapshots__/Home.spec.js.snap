// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home #render renders correctly 1`] = `
<div
  class="home"
  element-loading-background="rgba(0, 0, 0, 0.8)"
  element-loading-spinner="el-icon-loading"
  element-loading-text="Loading..."
>
  <h1
    class="home__title text-center"
  >
    
    Star Wars App
  
  </h1>
   
  <el-row-stub
    class="home__search"
  >
    <el-col-stub
      md="8"
      offset-sm="12"
      sm="12"
    >
      <el-input-stub
        placeholder="What do you want to search?"
        value=""
      >
        <el-button-stub
          icon="el-icon-search"
        />
      </el-input-stub>
    </el-col-stub>
  </el-row-stub>
   
  <el-row-stub
    gutter="20"
  >
     
    <el-col-stub
      span="24"
    >
      <!---->
    </el-col-stub>
  </el-row-stub>
   
  <!---->
</div>
`;
